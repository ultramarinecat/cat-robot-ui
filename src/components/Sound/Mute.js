//
// Mute button
// Updates icon based on muted state
//

import React, { memo } from 'react';
import PropTypes from 'prop-types';

import styles from './Mute.module.scss';

const MUTE_ICON = 'volume_down';
const UNMUTE_ICON = 'volume_mute';

export function Mute({ isMuted, onClick }) {
  const muteButtonIcon = isMuted ? UNMUTE_ICON : MUTE_ICON;

  return (
    <span onClick={onClick} data-testid="mute">
      <i
        className={`${styles.main} material-icons`}
        data-testid={isMuted ? 'unmute-icon' : 'mute-icon'}
      >
        {muteButtonIcon}
      </i>
    </span>
  );
}

Mute.propTypes = {
  isMuted: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default memo(Mute);
