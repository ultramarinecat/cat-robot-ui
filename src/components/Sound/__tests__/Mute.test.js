import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import { Mute } from '../Mute.js';

const MUTE = 'mute';
const MUTE_ICON = 'mute-icon';
const UNMUTE_ICON = 'unmute-icon';

describe('Mute', () => {
  it('should display unmute icon if `isMuted` is true', () => {
    const { getByTestId, queryByTestId, container } = render(
      <Mute isMuted={true} onClick={jest.fn()} />
    );

    const mute = getByTestId(MUTE);
    const muteIcon = queryByTestId(MUTE_ICON);
    const unmuteIcon = queryByTestId(UNMUTE_ICON);

    expect(mute).toBeInTheDocument();
    expect(unmuteIcon).toBeInTheDocument();
    expect(muteIcon).toBeNull();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display mute icon if `isMuted` is false', () => {
    const { getByTestId, queryByTestId, container } = render(
      <Mute isMuted={false} onClick={jest.fn()} />
    );

    const mute = getByTestId(MUTE);
    const muteIcon = queryByTestId(MUTE_ICON);
    const unmuteIcon = queryByTestId(UNMUTE_ICON);

    expect(mute).toBeInTheDocument();
    expect(muteIcon).toBeInTheDocument();
    expect(unmuteIcon).toBeNull();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call callback prop when clicked', () => {
    const handleClick = jest.fn();

    const { getByTestId } = render(<Mute isMuted={true} onClick={handleClick} />);

    const mute = getByTestId(MUTE);
    fireEvent.click(mute);

    expect(handleClick).toHaveBeenCalled();
  });
});
