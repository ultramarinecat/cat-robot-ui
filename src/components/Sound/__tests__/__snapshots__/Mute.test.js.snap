// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mute should display mute icon if \`isMuted\` is false 1`] = `
<span
  data-testid="mute"
>
  <i
    class="main material-icons"
    data-testid="mute-icon"
  >
    volume_down
  </i>
</span>
`;

exports[`Mute should display unmute icon if \`isMuted\` is true 1`] = `
<span
  data-testid="mute"
>
  <i
    class="main material-icons"
    data-testid="unmute-icon"
  >
    volume_mute
  </i>
</span>
`;
